src/vivarium_census_prl_synth_pop/utilities.py

```diff
@@ -385,3 +385,51 @@ def sample_acs_group_quarters(
     )
 
     return chosen_persons
+
+
+def randomly_sample_states_pumas(
+    unit_ids: pd.Series,
+    available_states: List[int],
+    additional_key: str,
+    randomness_stream: RandomnessStream = None,
+    random_seed: int = None,
+) -> pd.DataFrame:
+    """Randomly sample new states and pumas from the raw data file.
+
+    Args:
+        unit_ids (pd.Series): household_ids or business_ids to sample for
+        available_states (List[int]): states in the artifact data
+        additional_key (str): standard RandomnessStream additional_key
+        randomness_stream (RandomnessStream, optional): Defaults to None.
+        random_seed (int, optional): Only used when trying to call
+            vectorized_choice when no RandomnessStream is available
+            (eg during setup). Defaults to None.
+
+    Returns:
+        pd.DataFrame: Index are the unit_ids and has columns [['state_id', 'puma']]
+    """
+    if random_seed is None and randomness_stream is None:
+        raise RuntimeError("A randomness_stream or random_seed must be provided")
+    if random_seed is not None and randomness_stream is not None:
+        raise RuntimeError("Only one of randomness_stream or random_seed should be provided")
+
+    state_puma_options = pd.read_csv(paths.PUMA_TO_ZIP_DATA_PATH)[
```

Comment:
We don't want to have to re-read from this file every time we sample. We should pass the state/puma combination options as arguments.